plugins/out_stackdriver/stackdriver.c

```diff
@@ -264,7 +325,18 @@ static flb_sds_t get_google_token(struct flb_stackdriver *ctx)
     int ret = 0;
     flb_sds_t output = NULL;
 
-    if (pthread_mutex_lock(&ctx->token_mutex)){
+    ret = pthread_mutex_trylock(&ctx->token_mutex);
+    if (ret == EBUSY) {
+        /*
+         * If the routine is locked we just use our pre-cached values and
+         * compose the expected authorization value.
+         *
+         * If the routine fails it will return NULL and the caller will just
+         * issue a FLB_RETRY.
+         */
+        return oauth2_cache_to_token();
```

Comment:
Isn't the cached value effectively the same as `ctx->o`? If that one is expired, the agent will attempt to send data with a stale token, wouldn't it?